Spigot-Server-Patches/0415-Add-BaseComponent-to-a-variety-of-events.patch

```diff
@@ -0,0 +1,403 @@
++                    displayName = com.destroystokyo.paper.util.ChatComponentHelper.baseComponentFromVanilla(((CraftPlayer) player).getHandle().getDisplayName());
++                }
+                 if (this.player.getWorld().paperConfig.useVanillaScoreboardColoring) {
+-                    IChatBaseComponent nameFromTeam = ScoreboardTeam.a(this.player.getTeam(),((CraftPlayer) player).getHandle().getDisplayName());
++                    IChatBaseComponent nameFromTeam = ScoreboardTeam.a(this.player.getTeam(), com.destroystokyo.paper.util.ChatComponentHelper.vanillaFromBaseComponent(displayName));
+                     // Explicitly add a RESET here, vanilla uses components for this now...
+-                    displayName = CraftChatMessage.fromComponent(nameFromTeam, EnumChatFormat.WHITE) + org.bukkit.ChatColor.RESET;
++                    displayName = net.md_5.bungee.api.chat.TextComponent.fromLegacyText(CraftChatMessage.fromComponent(nameFromTeam, EnumChatFormat.WHITE) + org.bukkit.ChatColor.RESET);
+                 }
+ 
+-                s = String.format(event.getFormat(), displayName, event.getMessage());
++                final net.md_5.bungee.api.chat.BaseComponent[] message = com.destroystokyo.paper.utils.BaseComponentFormatter.format(event.getFormatComponents(), displayName, event.getMessageComponents()); // Paper - Add BaseComponent to events
+                 // Paper end
+-                minecraftServer.console.sendMessage(s);
++                minecraftServer.console.sendMessage(message); // Paper - Add BaseComponent to events
+                 if (((LazyPlayerSet) event.getRecipients()).isLazy()) {
+                     for (Object recipient : minecraftServer.getPlayerList().players) {
+-                        ((EntityPlayer) recipient).sendMessage(CraftChatMessage.fromString(s));
++                        ((EntityPlayer) recipient).getBukkitEntity().sendMessage(message); // Paper - Add BaseComponent to events
+                     }
+                 } else {
+                     for (Player recipient : event.getRecipients()) {
+-                        recipient.sendMessage(s);
++                        recipient.sendMessage(message); // Paper - Add BaseComponent to events
+                     }
+                 }
+             }
+diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
+index fbc8a33aa..be00991b2 100644
+--- a/src/main/java/net/minecraft/server/PlayerList.java
++++ b/src/main/java/net/minecraft/server/PlayerList.java
+@@ -177,7 +177,7 @@ public abstract class PlayerList {
+         }
+ 
+         chatmessage.a(EnumChatFormat.YELLOW);
+-        this.onPlayerJoin(entityplayer, CraftChatMessage.fromComponent(chatmessage, EnumChatFormat.WHITE)); // Paper
++        this.onPlayerJoin(entityplayer, com.destroystokyo.paper.util.ChatComponentHelper.baseComponentFromVanilla(chatmessage)); // Paper // Paper - Add BaseComponents to events
```

Comment:
need to review the potential consequences of removing this  CraftChatMessage.fromComponent call (and any others that appear in this patch)